kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java

```diff
@@ -442,27 +430,35 @@ private AmazonS3 getConnection(String bucketName) {
 		 String finalObjectName=null;
      	if(useAccountAsBucketname) {
      		 bucketName=account;
-     		 finalObjectName = ObjectStoreUtil.getName(container,null,TAGS_FILENAME);
+     		 finalObjectName = ObjectStoreUtil.getName(container,null,TAGS_FILENAME)+SEPARATOR;
      	}else {
      		 bucketName=container;
-     		 finalObjectName = TAGS_FILENAME;
+     		 finalObjectName = TAGS_FILENAME+SEPARATOR;
      	}
 		AmazonS3 connection = getConnection(bucketName);
-		if(connection.doesObjectExist(bucketName, finalObjectName)) {
 		
-		GetObjectTaggingRequest getObjectTaggingRequest=new GetObjectTaggingRequest(bucketName,finalObjectName);
-		
-			GetObjectTaggingResult getObjectTaggingResult=connection.getObjectTagging(getObjectTaggingRequest);
-		
-		if(getObjectTaggingResult!=null) {
-			List<Tag> tagSet = getObjectTaggingResult.getTagSet();
-			if (tagSet != null) {
-				for(Tag tag:tagSet) {
-					objectTags.put(tag.getKey(), tag.getValue());
-				}
-			}
-		  }
-		}
+		List<S3ObjectSummary> os = null;
+	   	   if(useAccountAsBucketname)
+	           os = connection.listObjects(bucketName, finalObjectName).getObjectSummaries();
+	   	   else
+	           os = connection.listObjects(bucketName).getObjectSummaries();
+
+	   	   List<String> tagNames=new ArrayList<String>();	   
+	   	if (os != null && os.size() > 0) {
+        
+            os.forEach(o -> {
+                String[] keys = o.getKey().split("/");
+                if (ArrayUtils.isNotEmpty(keys)) {
+                	tagNames.add(keys[2]);

```

Comment:
this will not work when the useAccountAsBucketname is false

Reply:
This will work as finalObjectName will be rid/tags/tagname

Reply:
In case of useAccountAsBucketname is false, will object objectSummary not be tags/tagname without the rid ?

Reply:
Fix : if(useAccountAsBucketname)
				objectSummary = connection.listObjects(bucketName, finalObjectName).getObjectSummaries();
	   	   else
				objectSummary = connection.listObjects(bucketName).getObjectSummaries();


if (useAccountAsBucketname) {
							if (keys[1] != null && keys[1].endsWith(TAGS_FILENAME))
								tagNames.add(keys[2]);
						} else {
							if (keys[0] != null && keys[0].endsWith(TAGS_FILENAME))
								tagNames.add(keys[1]);
						}